v2/api-validator/tests/server-tests/collateral/settlement.test.ts

```diff
@@ -4,20 +4,44 @@ import {
   SettlementInstructions,
   SettlementState,
   SettlementWithdrawInstruction,
+  NativeCryptocurrency,
+  OtherAssetReference,
+  NationalCurrency,
 } from '../../../src/client/generated';
 import { v4 as uuid } from 'uuid';
 import config from '../../../src/config';
 import Client from '../../../src/client';
 
 const noCollateralCapability = !hasCapability('collateral');
+const noTransferCapability = !hasCapability('transfers');
 
-describe.skipIf(noCollateralCapability)('Collateral Settlements', () => {
+describe.skipIf(noCollateralCapability || noTransferCapability)('Collateral Settlements', () => {
   const client: Client = new Client();
   const collateralId: string = config.get('collateral.collateralAccount.accountId');
   let accountId: string;
 
+  function isNativeCryptocurrency(
+    asset: NativeCryptocurrency | OtherAssetReference | NationalCurrency
+  ): asset is NativeCryptocurrency {
+    return typeof asset === 'object' && 'cryptocurrencySymbol' in asset && 'blockchain' in asset;
+  }
+
+  function isOtherAssetReference(
+    asset: NativeCryptocurrency | OtherAssetReference | NationalCurrency
+  ): asset is OtherAssetReference {
+    return typeof asset === 'object' && 'assetId' in asset;
+  }
+
   beforeAll(async () => {
     accountId = getCapableAccountId('collateral');
+
+    // Validating that withdrawal & deposit capabilities are enabled, as it is a must for collateral settlement operations.
+    expect(() => {
+      client.capabilities.getWithdrawalMethods({ accountId });
+    }).not.toThrow();
+    expect(() => {
+      client.capabilities.getDepositMethods({ accountId });
+    }).not.toThrow();
   });
 
   describe('Check full settlement flow', () => {
```

Comment:
All the tests in this block assume a very specific order of execution because only the first test sets `depositInstructions` and `withdrawInstructions`. Tests should never rely on data acquired in another test.